src/webapp/hooks/useUserNotifications.ts

```diff
@@ -0,0 +1,49 @@
+import { AppContextProviderProps } from "../contexts/app-context";
+import { useEffect, useState } from "react";
+import { UserNotificationDialogProps } from "../components/user-notification/UserNotificationDialog";
+
+export function useUserNotifications(props: useUserNotificationProps) {
+    const { appContextProps } = props;
+    const [isUserNotifsLoading, setIsUserNotifsLoading] = useState(true);
+    const [userNotificationDialogProps, setUserNotificationDialogProps] = useState<UserNotificationDialogProps | null>(
+        null
+    );
```

Comment:
No need to change, but typically initial empty values are handled with "undefined": `useState<UserNotificationDialogProps>()`